diff --git a/node_modules/braces/lib/parse.js b/node_modules/braces/lib/parse.js
index 3a6988e6..a152c33b 100644
--- a/node_modules/braces/lib/parse.js
+++ b/node_modules/braces/lib/parse.js
@@ -196,6 +196,9 @@ const parse = (input, options = {}) => {
 
     if (value === CHAR_LEFT_CURLY_BRACE) {
       depth++;
+      if (depth > (opts.maxDepth || 250)) {
+        throw new SyntaxError(`Exceeded maximum brace nesting depth (${opts.maxDepth || 250})`);
+      }
 
       const dollar = prev.value && prev.value.slice(-1) === '$' || block.dollar === true;
       const brace = {
